src/main/java/hanshyn/onlinebookstore/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,51 @@
+package hanshyn.onlinebookstore.repository;
+
+import hanshyn.onlinebookstore.model.Book;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```

Comment:
You can use @RequiredArgsConstructor() instead write constructor manually from library lombok